topic04-git-introduction/book-git-tutorial/02.Branching.md

```diff
@@ -0,0 +1,40 @@
+# Git Branching
+
+----
+## What is Branching?
+
+
+This document is to show how branching works in git and how to use it. Branching is a feature available in git that lets you pull off from the master branch to make changes that won't directly effect the master branch unless merged. It also means that others have to review your work first just in case you have errors in your work such as spelling or bugs.
+
+## Why Would You Use Branching
+
+The main reason using different branches is so that you are not pushing to master. Pushing to master won't show any history bar the commit. 
```

Comment:
Git history is the commits. The branch would still show history (commits). However, your first sentence would be the main point. We don't want to push anything to master directly to not affect it especially if you're experimenting or doing something big enough that would cause disruption on the master branch. The created branch would be a safe space for someone to experiment in and do major changes without disrupting the main branch.